src/app.js

```diff
@@ -51,48 +51,59 @@ const isIdNum = (req, res, next) => {
 }
 
 // checks object keys are valid 
-const checkObjKeys = (req, res, next) => { 
-    const possibleKeys = ["name", "sex", "coat", "description", "breedId"]; 
+const checkObjKeys = (req, res, next) => {  
+
+    const possibleKeys = ["name", "sex", "coat", "description", "breed"];
+
+    function isKeyPresent (key) { 
+        console.log(`is ${key} present?`)
+        return possibleKeys.includes(key)
+    }
+
     const objToCheck = req.query;                // { name: "", sex: "", coat: ""}
-    arrayOfKeys = Object.keys(objToCheck);      // Eg: [ "name", "sex", "coat"]
-    arrayOfKeys.forEach((key) => { 
-        console.log(`checking key: ${key}`)
-        const index = possibleKeys.indexOf(key);    // alternative is to use .findByIndex(callback)
-        if (index < 0) {
-            let message =  `"${key}" is not a valid property`;
-            const newError = new Error(message);
-            newError.status = 404;
-            return next(newError); 
-        } else {
-        console.log(`key '${key}' verified`)
-        }
-    });
-    console.log('object successfully checked');
-    req.object = objToCheck; 
-    next();
+    console.log(objToCheck);
+    const arrayOfKeys = Object.keys(objToCheck);      // Eg: [ "name", "sex", "coat"]
+    console.log(arrayOfKeys);
+
+    if (arrayOfKeys.length < 1) {
+        const newError = new Error('no parameters provided');
+        newError.status = 404;
+        return next(newError);
+    }
+
+    if (!arrayOfKeys.every(isKeyPresent)) {
+        console.log(`One or more property is invalid`);
+        const newError = new Error('One or more property is invalid');
```

Comment:
A friendlier error message for a user of the API might be to log which properties are invalid. But this is probably a low priority thing to address later on!

Reply:
It originally did, but the code got overly-complex and I simplified it ...as well as the functionality! Will look at again.